src/classes/trackers.js

```diff
@@ -1,290 +1,412 @@
+            tracker = this.findTracker(cnameRequestData)
+            if (tracker) {
+                requestData = cnameRequestData
+            } else {
+                if (firstParty) {
                     return null
                 }
-            }
-
-            // finds a matching rule by iterating over the rules in tracker.data and sets redirectUrl.
-            const matchedRule = this.findRule(tracker, requestData)
-
-            const redirectUrl = (matchedRule && matchedRule.surrogate) ? this.surrogateList[matchedRule.surrogate] : false
-
-            // sets tracker.exception by looking at tracker.rule exceptions (if any)
-            const matchedRuleException = matchedRule ? this.matchesRuleDefinition(matchedRule, 'exceptions', requestData) : false
-
-            const trackerOwner = this.findTrackerOwner(requestData.urlToCheckDomain)
-
-            const websiteOwner = this.findWebsiteOwner(requestData)
-
-            const firstParty = (trackerOwner && websiteOwner) ? trackerOwner === websiteOwner : false
-
-            const fullTrackerDomain = requestData.urlToCheckSplit.join('.')
-
-            const { action, reason } = this.getAction({
-                firstParty,
-                matchedRule,
-                matchedRuleException,
-                defaultAction: tracker.default,
-                redirectUrl
-            })
-
-            return {
-                action,
-                reason,
-                firstParty,
-                redirectUrl,
-                matchedRule,
-                matchedRuleException,
-                tracker,
-                fullTrackerDomain,
-                fromCname
-            }
-        }
-
-        /*
-         * Pull subdomains off of the reqeust rule and look for a matching tracker object in our data
-         */
-        findTracker (requestData) {
-            const urlList = Array.from(requestData.urlToCheckSplit)
-
-            while (urlList.length > 1) {
-                const trackerDomain = urlList.join('.')
-                urlList.shift()
-
-                const matchedTracker = this.trackerList[trackerDomain]
-                if (matchedTracker) {
-                    return matchedTracker
+                const owner = {
+                    name: requestOwner || requestData.urlToCheckDomain || 'unknown',
+                    displayName: requestOwner || requestData.urlToCheckDomain || 'Unknown'
+                }
+                /** @type {TrackerObj} */
+                const tracker = {
+                    domain: fullTrackerDomain,
+                    owner: owner,
+                    prevalence: 0,
+                    fingerprinting: 0,
+                    cookies: 0,
+                    categories: [],
+                    default: 'unknown',
+                    rules: []
+                }
+                return {
+                    action: tracker.default,
```

Comment:
nit: `action: unknown` sounds weird, like we don't know what to do? Maybe `action: none` or `action: nontracker`?

Reply:
Changed to none as this feels the most accurate; and yeah we kinda don't know what to do really.